core/jvm/src/test/scala/fs2/StreamSpec.scala

```diff
@@ -100,6 +100,17 @@ class StreamSpec extends Fs2Spec with Inside {
         IndexedSeq.range(0, 100)
     }
 
+    "repartition" in {
+      Stream("Lore", "m ip", "sum dolo", "r sit amet").repartition(s => Segment.array(s.split(" "))).toList shouldBe
+        List("Lorem", "ipsum", "dolor", "sit", "amet")
+      Stream("hel", "l", "o Wor", "ld").repartition(s => Segment.indexedSeq(s.grouped(2).toVector)).toList shouldBe
+        List("he", "ll", "o ", "Wo", "rl", "d")
+      Stream(1, 2, 3, 4, 5).repartition(i => Segment.indexedSeq(Vector(i, i))).toList shouldBe
+        List(1, 3, 6, 10, 15, 15)
+      (Stream(): Stream[Nothing, String]).repartition(_ => Segment.empty).toList shouldBe List()
```

Comment:
This can probably be `Stream.empty.covaryOutput[String]` or `Stream(): Stream[Pure, String]`

Reply:
Thanks! I think the surprising behavior you found was due to ending the pull too early. I've added some of those cases as test cases. Let me know if the behavior now makes sense.